src/saltext/vmware/modules/esxi.py

```diff
@@ -1166,6 +1166,246 @@ def get_user(
         raise salt.exceptions.SaltException(str(exc))
 
 
+def add_role(
+    role_name,
+    privilege_ids,
+    esxi_host_name=None,
+    esxi_user_name=None,
+    esxi_user_password=None,
```

Comment:
I'm confused, do the regular service instance credentials not work for setting roles (maybe the role needs to be set after directly connecting to the host)? 

If so, we might need to provide a structured way of having per-host credentials in the pillar data.

Reply:
The AuthorizationManager attribute is available only on the service instance and not at per host level like we did it for users. So, the roles need to be set at every host (pyvmomi shortcoming).

Managing the credentials of every host in the pillar will be a daunting task. We are using call level creds for tags and emulating the same pattern here. Prior discussion here: https://vmware.slack.com/archives/G01PF63FRRS/p1638901094043600

Reply:
We may need to have a different discussion about this - I'm suspicious that handling credentials in any way but pillar is a good idea.

Reply:
We are currently accepting creds in the calls for all REST APIs today (Tags + NSX). Need to streamline the way things are handled at other places. I will move the handling to pillars for host-wise creds for this PR.